code/modules/overmap/ftl_shunt/computer.dm

```diff
@@ -0,0 +1,150 @@
+/obj/machinery/computer/ship/ftl
+	name = "superluminal shunt control console"
+	icon_keyboard = "teleport_key"
+	icon_screen = "teleport"
+	light_color = "#77fff8"
+	var/obj/machinery/ftl_shunt/core/linked_core
+	var/cost = 0
+
+/obj/machinery/computer/ship/ftl/Initialize()
+	. = ..()
+	find_core()
+
+/obj/machinery/computer/ship/ftl/Destroy()
+	. = ..()
+	if(linked_core)
+		linked_core.ftl_computer = null
+		linked_core = null
+
+/obj/machinery/computer/ship/ftl/proc/recalc_cost()
+	var/jump_dist = get_dist(linked, locate(linked_core.shunt_x, linked_core.shunt_y, GLOB.using_map.overmap_z))
```

Comment:
Maybe return `INFINITY` or something if `linked` or `linked_core` are null?

Reply:
This proc should never run if either are linked, but I'll add a guard for it.